src/test/unit/lib/model/OutboundRequestToPayModel.test.js

```diff
@@ -0,0 +1,754 @@
+        MojaloopRequests.__getParties = jest.fn(() => Promise.resolve());
+        MojaloopRequests.__postTransactionRequests = jest.fn(() => Promise.resolve());
+        
+        cache = new Cache({
+            host: 'dummycachehost',
+            port: 1234,
+            logger,
+        });
+        await cache.connect();
+    });
+
+    afterEach(async () => {
+        await cache.disconnect();
+    });
+
+    test('initializes to starting state', async () => {
+        const model = new Model({
+            cache,
+            logger,
+            ...config,
+        });
+
+        await model.initialize(JSON.parse(JSON.stringify(requestToPayRequest)));
+        expect(StateMachine.__instance.state).toBe('start');
+    });
+
+
+    test('executes all two stages without halting when AUTO_ACCEPT_PARTY is true', async () => {
+        config.autoAcceptParty = true;
+        
+        MojaloopRequests.__getParties = jest.fn(() => {
+            emitPartyCacheMessage(cache, payeeParty);
+            return Promise.resolve();
+        });
+
+        MojaloopRequests.__postTransactionRequests = jest.fn((postTransactionRequestsBody) => {
+            // ensure that the `MojaloopRequests.postTransactionRequestsBody` method has been called with correct arguments
+            // including extension list
+            // const extensionList = postTransactionRequestsBody.extensionList;
+            // expect(extensionList).toBeTruthy();
+            // expect(extensionList.extension).toBeTruthy();
+            // expect(extensionList.extension.length).toBe(2);
+            // expect(extensionList.extension[0]).toEqual({ key: 'key1', value: 'value1' });
+            // expect(extensionList.extension[1]).toEqual({ key: 'key2', value: 'value2' });
+
+            // simulate a callback with the quote response
+            emitTransactionRequestResponseCacheMessage(cache, postTransactionRequestsBody.transactionRequestId, transactionRequestResponse);
+            return Promise.resolve();
+        });
+
+        const model = new Model({
+            cache,
+            logger,
+            ...config,
+        });
+
+        await model.initialize(JSON.parse(JSON.stringify(requestToPayRequest)));
+
+        expect(StateMachine.__instance.state).toBe('start');
+
+        // start the model running
+        const result = await model.run();
+
+        console.log(`Result after two stage transfer: ${util.inspect(result)}`);
+
+        expect(MojaloopRequests.__getParties).toHaveBeenCalledTimes(1);
+        expect(MojaloopRequests.__postTransactionRequests).toHaveBeenCalledTimes(1);
+        
+        // check we stopped at payeeResolved state
+        expect(result.currentState).toBe('COMPLETED');
+        expect(result.transactionRequestResponse.transactionRequestState).toBe('RECEIVED');
+        expect(StateMachine.__instance.state).toBe('succeeded');
+    });
+
+
+    // test('uses quote response transfer amount for transfer prepare', async () => {
```

Comment:
I think better to remove such code in comments from this repository and leave it in only in your repo (in a different branch) if necessary.
For this repository, you can put some TODOs.